src/components/pages/home/AddressLink.tsx

```diff
@@ -0,0 +1,52 @@
+import { useContext } from "react";
+import Link from "next/link";
+import { AddressContext } from "@/components/lib/context/AddressContext";
+
+interface AddressLinkProps {
+  type: string;
+  label: string;
+  address: string | null;
+  href: string;
+  children: React.ReactNode;
+}
+
+const AddressLink = ({
+  type,
+  label,
+  address,
+  href,
+  children,
+}: AddressLinkProps) => {
+  const { state, dispatch } = useContext(AddressContext);
+
+  const handleMouseEnter = () => {
+    dispatch({ type: "SET_HOVERED_ADDRESS", payload: address });
+  };
+
+  const handleMouseLeave = () => {
+    dispatch({ type: "SET_HOVERED_ADDRESS", payload: null });
+  };
+
+  const className = `truncate ${
```

Comment:
I think the truncate / max-width class names should be moved one level up and we should let the parent component decide how to handle the address styling.
Use an `addressClassName` prop instead and pass it to the underlying span displaying the address.